src/dimse/wadoRs.ts

```diff
@@ -1,72 +1,282 @@
 import { ConfParams, config } from '../utils/config';
 import { LoggerSingleton } from '../utils/logger';
-import { fileExists } from '../utils/fileHelper';
 import { waitOrFetchData } from './fetchData';
 import { compressFile } from './compressFile';
-import dicomParser from 'dicom-parser';
-import crypto from 'crypto';
+import { doFind } from '../dimse/findData';
 import path from 'path';
-import fs from 'fs';
+import fs from 'fs/promises';
 import { QUERY_LEVEL } from './querLevel';
+import deepmerge from 'deepmerge';
+import dicomParser from 'dicom-parser';
+import combineMerge from '../utils/combineMerge';
+import { fileExists } from '../utils/fileHelper';
+import { execFile as exFile } from 'child_process';
+import util from 'util';
+
+const execFile = util.promisify(exFile);
+
+export type DataFormat = 'pixeldata' | 'bulkdata' | 'rendered' | 'thumbnail'
 
 type WadoRsArgs = {
   studyInstanceUid: string;
-  seriesInstanceUid: string;
-  sopInstanceUid: string;
+  seriesInstanceUid?: string;
+  sopInstanceUid?: string;
+  dataFormat?: DataFormat;
+  frame?: number | number[];
 };
 type WadoRsResponse = {
   contentType: string;
   buffer: Buffer;
 };
-export async function doWadoRs({ studyInstanceUid, seriesInstanceUid, sopInstanceUid }: WadoRsArgs): Promise<WadoRsResponse> {
+type QidoResponse = {
+  [key: string]: {
+    Value: string[]
+    vr: string
+  }
+}
+
+const term = '\r\n';
+
+/**
+ * This function uses DCMTK to convert the given DICOM file to JPEG format
+ * It will try twice, once without frames, the other with all frames.
+ * If both fail it will throw the error returned by DCMTK.
+ * If no resulting JPEG can be found, then it will return undefined.
+ * 
+ * @param filepath Path to the file to convert
+ * @param [asThumbnail=false] Return as thumbnail
+ */
+async function convertToJpeg(filepath: string, asThumbnail = false) {
```

Comment:
This requires a binary (compiled for the current platform) of DCMTK, which is against the goal of the project (not having to cope with this but rather rely on node-native dependencies for it. So currently dicom-dimse-native does not include the dicom to jpeg converter, but I would take this as a good opportunity to add it. Then we can replace this part with a native call.